pkg/messaging/v1/replayable_request.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
```

Comment:
Reviewers: this is a new base class for InvokeMethodRequest/InvokeMethodResponse.

Its job is to maintain a buffer where the entire request or response body is kept in memory if needed. Because streams can only be consumed once, if Resiliency is enabled we need to be able to replay requests/responses, which requires being able to keep the data in-memory too (while we send it to the consumer at the same time).